### PR TITLE
pnpm workspaces with typescript and cypress

### PR DESCRIPTION
**Pull request by admmasters**
## Documentation / Examples

Adding a new example showing how pnpm workspaces can work with nextjs - without going too crazy on tooling and just using pure pnpm workspaces.

- [x] Make sure the linting passes

### CODE REVIEW

**Comment by roim** [comment]
Would be good to also cover developer experience (eg tests, dev mode, etc).

We're using a similar setup at [Statsig](www.statsig.com) and found that tsc [Project References](https://www.typescriptlang.org/docs/handbook/project-references.html) work really well with pnpm and nextjs. E.g. for `pnpm run dev`, you can run tsc in watch mode before calling `next dev`.

**Comment by admmasters** [comment]
Good plan @roim will investigate that as an improvement.

File: examples/with-pnpm-workspaces/packages/web-app/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "compilerOptions": {
+    "target": "es5",
+    "lib": [
+      "dom",
+      "dom.iterable",
+      "esnext"
+    ],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "strict": false,
```
Comment (nhducit) [inline examples/with-pnpm-workspaces/packages/web-app/tsconfig.json:11]: should we enable Typescript strict mode?
```suggestion
    "strict": true,
```

File: examples/with-pnpm-workspaces/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Pnpm workspaces example
+
+[Pnpm](ttps://pnpm.io) has built-in support for monorepositories (AKA multi-package repositories, multi-project repositories, or monolithic repositories). You can create a workspace to unite multiple projects inside a single repository.
```
Comment (Ethan-Arrowood) [inline examples/with-pnpm-workspaces/README.md:3]:
```suggestion
[pnpm](https://pnpm.io) has built-in support for monorepositories (AKA multi-package repositories, multi-project repositories, or monolithic repositories). You can create a workspace to unite multiple projects inside a single repository.
```

File: examples/with-pnpm-workspaces/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+See: <https://pnpm.io/workspaces>
+
+## Why use it?
+
+If you want to get started quickly with a monorepo web-app utilising the excellent Pnpm's inbuilt workspaces feature.
```
Comment (Ethan-Arrowood) [inline examples/with-pnpm-workspaces/README.md:9]:
```suggestion
If you want to get started quickly with a monorepo web-app utilizing the excellent pnpm's built in workspaces feature.
```

**Review by Ethan-Arrowood — COMMENTED** [review]
Great work! Left some suggestions for you to consider 😁

**Review by ijjk — COMMENTED** [review]
Hi, I don't think we want to add another example for `cypress` per the example guidelines https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples feel free to update the existing example in a fresh PR though!

**Pull request closed without merging**